src/main.cpp

```diff
@@ -6289,10 +6289,11 @@ bool ProcessMessages(CNode* pfrom, CConnman& connman)
         uint256 hash = Hash(vRecv.begin(), vRecv.begin() + nMessageSize);
         if (memcmp(hash.begin(), hdr.pchChecksum, CMessageHeader::CHECKSUM_SIZE) != 0)
         {
-            LogPrintf("%s(%s, %u bytes): CHECKSUM ERROR expected %s was %s\n", __func__,
+            LogPrintf("%s(%s, %u bytes): CHECKSUM ERROR expected=%s was=%s peer=%d\n", __func__,
                SanitizeString(strCommand), nMessageSize,
+               HexStr(hdr.pchChecksum, hdr.pchChecksum+CMessageHeader::CHECKSUM_SIZE),
```

Comment:
No, you are correct of course. As `hash` is computed locally a few lines above, it is the expected value, thus the first one and the current code is OK.

But 'was' is strange here (it *was* received). There is similar code a few lines above:
```
%x expected, %x received
```